x-pack/plugins/cases/public/components/user_actions/types.ts

```diff
@@ -52,7 +53,7 @@ export interface UserActionBuilderArgs {
   selectedOutlineCommentId: string;
   loadingCommentIds: string[];
   loadingAlertData: boolean;
-  alertData: Record<string, Ecs>;
+  alertData: Record<string, unknown>;
```

Comment:
is `unknown` really the best type here? is there a closer real type we can use for this? like a Record or a string?

Reply:
I'm open to other ideas, but I'm not quite sure what to do since security solution's implementation of this hook used to return an `Ecs` object with the fields broken done into nested objects like:

```
{
  kibana: {
    alert: "some value",
  }
}
```

Where as this function will return it as:

```
{
  "kibana.alert": "some value"
}
```

So the values could be `object | string | string[] | number | number[]` and potentially nested

By putting `unknown` I think we force the recipient to figure out what type the data is before they use it which might be safe.

Reply:
Let's roll with this one and create an issue to work on a proper data type?

Reply:
Will do, I'll create an issue

Reply:
Issue: https://github.com/elastic/kibana/issues/124048